velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -291,38 +287,173 @@ TpchPlan TpchQueryBuilder::getQ6Plan() const {
+
+  std::string lineitemReturnFlagFilter = "l_returnflag = 'R'";
+  const auto orderDate = "o_orderdate";
+  std::string orderDateFilter;
+  // DWRF does not support Date type. Use Varchar instead.
+  if (ordersSelectedRowType->findChild(orderDate)->isVarchar()) {
+    orderDateFilter = "o_orderdate between '1993-10-01' and '1993-12-31'";
+  } else {
+    orderDateFilter =
+        "o_orderdate between '1993-10-01'::DATE and '1993-12-31'::DATE";
+  }
+
+  std::vector<std::string> customerJoinOutputColumns = {
+      "c_name", "c_acctbal", "c_phone", "c_address", "c_custkey", "c_comment"};
+
+  auto mergeColumnNames = [](std::vector<std::string>& v1,
+                             std::vector<std::string> v2) {
+    v1.insert(v1.end(), v2.begin(), v2.end());
+    return v1;
+  };
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId customerScanNodeId;
+  core::PlanNodeId nationScanNodeId;
+  core::PlanNodeId lineitemScanNodeId;
+  core::PlanNodeId ordersScanNodeId;
+
+  auto nation =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kNation_, nationSelectedRowType, nationFileColumns)
+          .capturePlanNodeId(nationScanNodeId)
+          .planNode();
+
+  auto orders = PlanBuilder(planNodeIdGenerator)
+                    .tableScan(
+                        kOrders_,
+                        ordersSelectedRowType,
+                        ordersFileColumns,
+                        {orderDateFilter})
+                    .capturePlanNodeId(ordersScanNodeId)
+                    .planNode();
+
+  auto partialPlan =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kCustomer_, customerSelectedRowType, customerFileColumns)
+          .capturePlanNodeId(customerScanNodeId)
+          .hashJoin(
+              {"c_custkey"},
+              {"o_custkey"},
+              orders,
+              "",
+              mergeColumnNames(
+                  customerJoinOutputColumns, {"c_nationkey", "o_orderkey"}))
+          .hashJoin(
+              {"c_nationkey"},
+              {"n_nationkey"},
+              nation,
+              "",
+              mergeColumnNames(
+                  customerJoinOutputColumns, {"n_name", "o_orderkey"}))
+          .planNode();
+
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(
+              kLineitem_,
+              lineitemSelectedRowType,
+              lineitemFileColumns,
+              {lineitemReturnFlagFilter})
+          .capturePlanNodeId(lineitemScanNodeId)
+          .project(
+              {"l_extendedprice * (1.0 - l_discount) AS part_revenue",
+               "l_orderkey"})
+          .hashJoin(
+              {"l_orderkey"},
+              {"o_orderkey"},
+              partialPlan,
+              "",
+              mergeColumnNames(
+                  customerJoinOutputColumns, {"part_revenue", "n_name"}))
+          .partialAggregation(
+              {"c_custkey",
```

Comment:
This list is also the customer columns + n_name. I don't think order is important for GROUP BY for partialAgg.

Reply:
Because we are using sortingKeys, the order of columns matters here.